Mathlib/SetTheory/Ordinal/FixedPoint.lean

```diff
@@ -48,44 +48,48 @@ finitely many functions in the family to `a`.
 
 `Ordinal.nfpFamily_fp` shows this is a fixed point, `Ordinal.le_nfpFamily` shows it's at
 least `a`, and `Ordinal.nfpFamily_le_fp` shows this is the least ordinal with these properties. -/
-def nfpFamily (f : ι → Ordinal → Ordinal) (a : Ordinal) : Ordinal :=
-  sup (List.foldr f a)
+def nfpFamily (f : ι → Ordinal.{max u v} → Ordinal.{max u v}) (a : Ordinal.{max u v}) : Ordinal :=
+  iSup (List.foldr f a)
 
 theorem nfpFamily_eq_sup (f : ι → Ordinal.{max u v} → Ordinal.{max u v}) (a : Ordinal.{max u v}) :
-    nfpFamily.{u, v} f a = sup.{u, v} (List.foldr f a) :=
+    nfpFamily.{u, v} f a = iSup (List.foldr f a) :=
   rfl
 
 theorem foldr_le_nfpFamily (f : ι → Ordinal → Ordinal)
     (a l) : List.foldr f a l ≤ nfpFamily.{u, v} f a :=
-  le_sup.{u, v} _ _
+  Ordinal.le_iSup _ _
 
 theorem le_nfpFamily (f : ι → Ordinal → Ordinal) (a) : a ≤ nfpFamily f a :=
-  le_sup _ []
+  Ordinal.le_iSup _ []
 
 theorem lt_nfpFamily {a b} : a < nfpFamily.{u, v} f b ↔ ∃ l, a < List.foldr f b l :=
-  lt_sup.{u, v}
+  Ordinal.lt_iSup
 
 theorem nfpFamily_le_iff {a b} : nfpFamily.{u, v} f a ≤ b ↔ ∀ l, List.foldr f a l ≤ b :=
-  sup_le_iff
+  Ordinal.iSup_le_iff
 
 theorem nfpFamily_le {a b} : (∀ l, List.foldr f a l ≤ b) → nfpFamily.{u, v} f a ≤ b :=
-  sup_le.{u, v}
+  Ordinal.iSup_le
 
-theorem nfpFamily_monotone (hf : ∀ i, Monotone (f i)) : Monotone (nfpFamily.{u, v} f) :=
-  fun _ _ h => sup_le.{u, v} fun l => (List.foldr_monotone hf l h).trans (le_sup.{u, v} _ l)
+theorem nfpFamily_monotone (hf : ∀ i, Monotone (f i)) : Monotone (nfpFamily.{u, v} f) := by
+  intro _ _ h
+  apply Ordinal.iSup_le
+  intro l
+  exact (List.foldr_monotone hf l h).trans (Ordinal.le_iSup _ l)
 
 theorem apply_lt_nfpFamily (H : ∀ i, IsNormal (f i)) {a b} (hb : b < nfpFamily.{u, v} f a) (i) :
     f i b < nfpFamily.{u, v} f a :=
   let ⟨l, hl⟩ := lt_nfpFamily.1 hb
-  lt_sup.2 ⟨i::l, (H i).strictMono hl⟩
+  Ordinal.lt_iSup.2 ⟨i::l, (H i).strictMono hl⟩
 
 theorem apply_lt_nfpFamily_iff [Nonempty ι] (H : ∀ i, IsNormal (f i)) {a b} :
-    (∀ i, f i b < nfpFamily.{u, v} f a) ↔ b < nfpFamily.{u, v} f a :=
-  ⟨fun h =>
-    lt_nfpFamily.2 <|
-      let ⟨l, hl⟩ := lt_sup.1 <| h <| Classical.arbitrary ι
-      ⟨l, ((H _).self_le b).trans_lt hl⟩,
-    apply_lt_nfpFamily H⟩
+    (∀ i, f i b < nfpFamily.{u, v} f a) ↔ b < nfpFamily.{u, v} f a := by
+  constructor
+  · intro h
+    exact lt_nfpFamily.2 <|
+      let ⟨l, hl⟩ := Ordinal.lt_iSup.1 <| h <| Classical.arbitrary ι
+      ⟨l, ((H _).self_le b).trans_lt hl⟩
+  · exact apply_lt_nfpFamily H
```

Comment:
```suggestion
    (∀ i, f i b < nfpFamily.{u, v} f a) ↔ b < nfpFamily.{u, v} f a where
  mp h :=
    lt_nfpFamily.2 <|
      let ⟨l, hl⟩ := Ordinal.lt_iSup.1 <| h <| Classical.arbitrary ι
      ⟨l, ((H _).self_le b).trans_lt hl⟩
  mpr := apply_lt_nfpFamily H
```
if you like it

Reply:
It looks nice, but I prefer the original. I wouldn't usually think of `Iff` as a structure to use `where` notation with.

Reply:
FYI, proofs of `Iff` using `where` are getting pretty common now.

Reply:
Someone else can change it later down the line :P